packages/support/src/Actions/Concerns/HasLifecycleHooks.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Filament\Support\Actions\Concerns;
+
+use Closure;
+
+trait HasLifecycleHooks
+{
+    protected ?Closure $before = null;
+
+    protected ?Closure $after = null;
+
+    protected ?Closure $beforeFormFilled = null;
+
+    protected ?Closure $afterFormFilled = null;
+
+    protected ?Closure $beforeFormValidated = null;
+
+    protected ?Closure $afterFormValidated = null;
```

Comment:
I know these are "LifecycleHooks", but shouldn't these be move to `HasFormSchema` as there is no use for them without a form?

Reply:
not sure, I think I want to keep all hooks in the same place tbh